packages/next/client/image.tsx

```diff
@@ -243,11 +245,11 @@ export type ImageProps = Omit<
   'src' | 'srcSet' | 'ref' | 'width' | 'height' | 'loading'
 > & {
   src: string | StaticImport
-  width?: number | string
-  height?: number | string
+  width?: NumericValue
+  height?: NumericValue
   layout?: LayoutValue
   loader?: ImageLoader
-  quality?: number | string
+  quality?: NumericValue
```

Comment:
The changes to `next/image` and `next/future/image` look good. Feel free to move that in a separate PR so we can merge it in.

Also add a test to invalid.tsx and valid.tsx fixture here: https://github.com/vercel/next.js/tree/canary/test/integration/image-future/typescript/pages

Reply:
Cheers, have added some tests there!